pyteal/ast/abi/tuple.py

```diff
@@ -117,6 +118,99 @@ def _encode_tuple(values: Sequence[BaseType]) -> Expr:
     return Concat(*toConcat)
 
 
+def _index_tuple_bytes(
```

Comment:
If at all possible it would be preferable not to maintain tuple indexing code in two separate places. Can the majority of the complexity duplicated in `_index_tuple_bytes` and `_index_tuple` be shared somewhere? Maybe there should be a helper function which does the heavy lifting for these?

Reply:
I also attempted to unify the largely same logic into a same function by https://github.com/algorand/pyteal/pull/629, please take a look?